site/app/libraries/database/PostgresqlDatabaseQueries.php

```diff
@@ -825,5 +829,305 @@ public function updateLateDays($user_id, $timestamp, $days, $csv_option=null) {
+              gcd.array_graded_version,
+              gcd.array_grade_time,
+              gcd.array_mark_id,
+              egd.array_version,
+              egd.array_non_hidden_non_extra_credit,
+              egd.array_non_hidden_extra_credit,
+              egd.array_hidden_non_extra_credit,
+              egd.array_hidden_extra_credit,
+              egd.array_submission_time,
+              egd.array_autograding_complete,
+              egv.active_version
+            FROM gradeable_data gd
+              LEFT JOIN (
+                SELECT
+                  array_agg(in_gcd.gc_id) as array_comp_id,
+                  array_agg(gcd_score) as array_score,
+                  array_agg(gcd_component_comment) as array_comment,
+                  array_agg(gcd_grader_id) as array_grader_id,
+                  array_agg(gcd_graded_version) as array_graded_version,
+                  array_agg(gcd_grade_time) as array_grade_time,
+                  array_agg(string_mark_id) as array_mark_id,
+                  in_gcd.gd_id
+                FROM gradeable_component_data in_gcd
+                  LEFT JOIN (
+                    SELECT
+                      array_to_string(array_agg(gcm_id), ',') AS string_mark_id,
+                      gc_id,
+                      gd_id
+                    FROM gradeable_component_mark_data
+                    GROUP BY gc_id, gd_id
+                  ) AS gcmd ON gcmd.gc_id=in_gcd.gc_id AND gcmd.gd_id=in_gcd.gd_id
+                GROUP BY in_gcd.gd_id
+              ) AS gcd ON gcd.gd_id=gd.gd_id
+              LEFT JOIN (
+                SELECT * 
+                FROM electronic_gradeable_version
+              ) AS egv ON (egv.team_id=gd.gd_team_id OR egv.user_id=gd.gd_user_id) AND egv.g_id=gd.g_id
+              LEFT JOIN (
+                SELECT
+                  array_agg(in_egd.g_version) AS array_version,
+                  array_agg(in_egd.autograding_non_hidden_non_extra_credit) AS array_non_hidden_non_extra_credit,
+                  array_agg(in_egd.autograding_non_hidden_extra_credit) AS array_non_hidden_extra_credit,
+                  array_agg(in_egd.autograding_hidden_non_extra_credit) AS array_hidden_non_extra_credit,
+                  array_agg(in_egd.autograding_hidden_extra_credit) AS array_hidden_extra_credit,
+                  array_agg(in_egd.submission_time) AS array_submission_time,
+                  array_agg(in_egd.autograding_complete) AS array_autograding_complete,
+                  g_id,
+                  user_id,
+                  team_id
+                FROM electronic_gradeable_data as in_egd
+                GROUP BY g_id, user_id, team_id
+              ) AS egd ON (egd.team_id=gd.gd_team_id OR egd.user_id=gd.gd_user_id) AND egd.g_id=gd.g_id
+            WHERE gd.g_id=? AND ((gd.gd_user_id IN (?)) OR (gd.gd_team_id IN (?)) OR ?)";
+
+        $this->course_db->query($query, array(
+            $gradeable->getId(),
+            implode(',', $users),
```

Comment:
Does this paradigm work in PDO? I'd think that in executing your query, it'd only quote around the whole string (so it'd be doing `gd.gd_user_id IN ('user_1, user_2')` instead of `gd.gd_user_id IN ('user_1', 'user_2')`